File: algorithms/binary_search/iterative.js
/* eslint-disable require-jsdoc */

export function binarySearch(array, item) {
  let left;
  let mid;
  let right;

  const last = array.length - 1;
  if (item < array[0] || item > array[last]) {
    return false;
  }

  left = 0;
  right = last;

  while (left <= right) {
    mid = parseInt((left + right) / 2);
    if (item == array[mid]) {
      return true;
    }

    if (item < array[mid]) {
      right = mid - 1;
    } else {
      left = mid + 1;
    }
  }

  return false;
}

File: algorithms/searching/binary-search/iterative.go
package binary_search

func BinarySearchIterative(array []int, item int) bool {
	if item < array[0] || item > array[len(array)-1] {
		return false
	}

	left := 0
	right := len(array) - 1

	for left <= right {
		mid := (left + right) / 2
		if array[mid] == item {
			return true
		}
		if item < array[mid] {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}

	return false
}
